### ISSUE TITLE
RuntimeError: CUDA error: no kernel image is available for execution on the device 

### ISSUE THREAD

**Issue by naykun** [pytorch/pytorch#35373]
### Please note that this issue tracker is not a help form and this issue will be closed.

We have a set of [listed resources available on the website](https://pytorch.org/resources). Our primary means of support is our discussion forum:

- [Discussion Forum](https://discuss.pytorch.org/)

Hi, I encountered an error when running my project using Pytorch 1.4 CUDA 10.1 on a Tesla V100 32GB GPU, any idea about why it happened? The environment is a docker container inherited from pytorch/pytorch:1.4-cuda10.1-cudnn7-devel, more specifically, I pushed my edited image here naykun/vrd:torch1.4-cuda10.1-apex-lsvrd-bert-stable.

![image](https://user-images.githubusercontent.com/20919663/77522760-b86a3b00-6ebf-11ea-9bd3-5e0aef091422.png)

![image](https://user-images.githubusercontent.com/20919663/77522791-c750ed80-6ebf-11ea-9a66-9cbb49bd19a7.png)

![image](https://user-images.githubusercontent.com/20919663/77522844-db94ea80-6ebf-11ea-9723-c7f5651a0515.png)
 

cc @ezyang @seemethere

**Comment by naykun** [comment]
And additional, this image and script work perfectly when I using a Tesla P100 GPU. But when I change to a Tesla V100 GPU, the above problem emerged.

**Comment by ngimel** [comment]
Normally this error means that code is not compiled for your GPU architecture, but our binaries are compiled with V100 support, so I'm not sure what's going on here.

**Comment by naykun** [comment]
I'm wondering whether this pytorch:1.4-cuda10.1-cudnn7-devel's binary supports half-precision computation on V100? Cuz I'm using apex during the training process, may that be the possible reason?

**Comment by ngimel** [comment]
It definitely supports half precision computation on V100. cc @ptrblck

**Comment by ptrblck** [comment]
As @ngimel explained, the binaries support halt precision.

Could you post a minimal code snippet to reproduce the error with the pytorch container, please?

**Comment by naykun** [comment]
I have just found the reason for the problem! It was caused by a wrong default setting during Apex compiling process. As described [here](https://github.com/NVIDIA/apex/issues/605#issuecomment-554453001) ,
And after manually set the target architecture, then re-install apex like below, problems will be solved.
```
$ export TORCH_CUDA_ARCH_LIST="6.0;6.1;6.2;7.0;7.5"
$ pip install -v --user --no-cache-dir --global-option="--cpp_ext" --global-option="--cuda_ext" ./
```
Hope this will help others. And by the way, much appreciated for your help! @ngimel @ptrblck
